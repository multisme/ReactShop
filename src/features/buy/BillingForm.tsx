import React from "react";
import { Link, useHistory } from "react-router-dom";

import { useFormFields } from "utils/utils";

const BillingForm = () => {
  const [fields, handleFieldChange] = useFormFields({
    Name: {},
    email: {},
    address: {},
    city: {},
    postalCode: {},
  });
  const history = useHistory();

  const handleSubmit = (e: any) => {
    e.preventDefault();
    history.push("/ship", [fields]);
  };

  return (
    <div id="BillingForm" className={"rectangle"} onSubmit={handleSubmit}>
      <form>
        <label>
          Billing Name:{" "}
          <input type="text" onChange={handleFieldChange} id="Name" required />
        </label>
        <label>
          Email:{" "}
          <input
            type="email"
            onChange={handleFieldChange}
            id="email"
            required
          />
        </label>
        <label>
          Billing Address:{" "}
          <input
            type="text"
            onChange={handleFieldChange}
            id="address"
            required
          />{" "}
        </label>
        <label>
          City:{" "}
          <input type="text" onChange={handleFieldChange} id="city" required />
        </label>
        <label>
          Postal Code:{" "}
          <input
            type="text"
            onChange={handleFieldChange}
            id="postalCode"
            required
          />
        </label>
        <fieldset className="button">
          <button type="submit">BILL</button>
          <Link to="/home">
            <button>CANCEL</button>
          </Link>
        </fieldset>
      </form>
    </div>
  );
};

export default BillingForm;
