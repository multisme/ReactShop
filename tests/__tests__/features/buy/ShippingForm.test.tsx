import BillingForm from "features/buy/BillingForm"
import {shallow} from "enzyme"

describe("BillingForm", () =>{
        it("rendets", () => {
                const component = shallow(<BillingForm />);
                expect(component).toMatchSnapshot()
        })
})
