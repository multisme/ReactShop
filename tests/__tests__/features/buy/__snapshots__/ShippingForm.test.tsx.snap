// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillingForm rendets 1`] = `
<div
  className="rectangle"
  id="BillingForm"
  onSubmit={[Function]}
>
  <form>
    <label>
      Billing Name:
       
      <input
        id="Name"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Email:
       
      <input
        id="email"
        onChange={[Function]}
        required={true}
        type="email"
      />
    </label>
    <label>
      Billing Address:
       
      <input
        id="address"
        onChange={[Function]}
        required={true}
        type="text"
      />
       
    </label>
    <label>
      City:
       
      <input
        id="city"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Postal Code:
       
      <input
        id="postalCode"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <fieldset
      className="button"
    >
      <button
        type="submit"
      >
        BILL
      </button>
      <Link
        to="/home"
      >
        <button>
          CANCEL
        </button>
      </Link>
    </fieldset>
  </form>
</div>
`;
